### ISSUE TITLE
Android: story titles image preview using wrong image

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1297]
User emailed to say that this feed is causing the wrong image to show. The web and iOS use a different image.

http://s1.dlnws.com/dealnews/rss/todays-edition.xml

![Screenshot_20190829-214228_NewsBlur](https://user-images.githubusercontent.com/44229/78706065-73410280-78dc-11ea-9a5f-27f1ef27d54c.jpg)

**Comment by TheQwerty** [comment]
FWIW, I've seen similar behavior from the [BoingBoing](http://feeds.feedburner.com/boingboing/) feed on Newsblur.com.
Given the repeated images I wondered if it might have something to do with stories that don't contain images causing some confusion or it grabbing images that are not part of the story (related stores, etc.).

Unfortunately I haven't been able to investigate it further than my own guesses though - sorry!

![image](https://user-images.githubusercontent.com/213205/94601046-e52d4f80-0260-11eb-8b07-6748436195f9.png)

**Comment by sictiru** [comment]
Looked at a few feeds and I attached some photos with the examples I found. (Boing Boing is a good example to replicate)
On the front end it tries to extract the story's thumbnail photo from the content and if that is not found, then it always uses the first entry from `story.image_urls`.

In the second photo, those stories return the same thumbnail url to the list for rendering.

In the forth photo, the expected thumbnail url is present in `story_image_urls` but as the second entry.


![IMG_20201004_133257](https://user-images.githubusercontent.com/13900221/95026899-6a41ab80-0649-11eb-8352-b4e92343a619.jpg)
![IMG_20201004_130632](https://user-images.githubusercontent.com/13900221/95026909-7a598b00-0649-11eb-876c-867e2228c349.jpg)
![IMG_20201004_132007](https://user-images.githubusercontent.com/13900221/95026918-8ba29780-0649-11eb-9cba-bd187e91c67e.jpg)
![IMG_20201004_141155](https://user-images.githubusercontent.com/13900221/95027219-ba217200-064b-11eb-9b29-1f2dd2048f5d.jpg)
